### PR TITLE
Run IWYU 20200408

### PR DESCRIPTION
**Pull request by jbytheway**
#### Summary
SUMMARY: None

#### Purpose of change
To keep our headers tidy.

#### Describe the solution
Run IWYU, clang-tidy, and IWYU again.

Add a script intended to help people do this in the future, since it's getting a bit complicated.

#### Describe alternatives you've considered
One change from before is now I'm using IWYU's `--reorder` feature to sort headers alphabetically.  Could have not done that.

#### Testing
Just verified that it builds.

#### Additional context
There's a reasonable chance of breaking the Windows or Android builds here.

### CODE REVIEW

File: src/sdl_wrappers.h [hunk 1]
```diff
@@ -5,17 +5,21 @@
...
 #   include <SDL.h>
-#   include <SDL_ttf.h>
 #   include <SDL_image.h>
+#   include <SDL_ttf.h>
 #endif
 // IWYU pragma: end_exports
 
 #include <memory>
 
+#include "SDL.h"
```
Comment (ZhilkinSerg) [inline src/sdl_wrappers.h:19]:
```suggestion
```

File: src/sdl_wrappers.h [hunk 2]
```diff
@@ -5,17 +5,21 @@
...
+#include "SDL_image.h"
```
Comment (ZhilkinSerg) [inline src/sdl_wrappers.h:20]:
```suggestion
```

File: src/sdl_wrappers.h [hunk 3]
```diff
@@ -5,17 +5,21 @@
...
+#include "SDL_ttf.h"
```
Comment (ZhilkinSerg) [inline src/sdl_wrappers.h:21]:
```suggestion
```

File: src/sdl_wrappers.h [hunk 4]
```diff
@@ -5,17 +5,21 @@
...
+
```
Comment (ZhilkinSerg) [inline src/sdl_wrappers.h:22]:
```suggestion
```

File: src/sdl_wrappers.h (hunk 1 above)
Comment (jbytheway) [reply src/sdl_wrappers.h:19]: Good point.  These SDL headers give the most grief!  Will have to look into why they're not being handled correctly any more...

**Comment by jbytheway** [comment]
Thanks.  Fixed that manually for now, but that looks like yet another IWYU bug I might have to look at at some point (probably one I introduced, this time...).

File: src/ime.cpp [hunk 5]
```diff
@@ -1,13 +1,9 @@
 #include "ime.h"
 
-#include "options.h"
-#include "platform_win.h"
-#include "sdltiles.h"
-
 #ifdef _WIN32
 
-#include <windows.h>
```
Comment (ZhilkinSerg) [inline src/ime.cpp:9]: I think this one should not be reordered and included before `imm.h`.

Comment (ZhilkinSerg) [reply src/ime.cpp:9]: Or maybe it was "platform_win.h".

File: src/ime.cpp [hunk 6]
```diff
@@ -1,13 +1,9 @@
 #include "ime.h"
 
-#include "options.h"
-#include "platform_win.h"
-#include "sdltiles.h"
```
Comment (ZhilkinSerg) [inline src/ime.cpp:5]: It is likely that this include is required for Android builds.

File: src/ime.cpp [hunk 7]
```diff
@@ -1,13 +1,9 @@
 #include "ime.h"
 
-#include "options.h"
```
Comment (ZhilkinSerg) [inline src/ime.cpp:3]: It is likely that this include is required for Android builds.

File: src/mapsharing.cpp [hunk 8]
```diff
@@ -1,18 +1,14 @@
 #include "mapsharing.h"
 
-#include "cata_utility.h"
-#include "filesystem.h"
-#include "platform_win.h"
```
Comment (ZhilkinSerg) [inline src/mapsharing.cpp:5]: This one is probably needed for GetCurrentProcessId in MinGW cross-compile.

File: src/translations.cpp [hunk 9]
```diff
@@ -42,12 +45,12 @@ static void reload_names()
...
 
 #if defined(LOCALIZE)
-#include "options.h"
 #include "debug.h"
+#include "options.h"
 #include "ui.h"
 #if defined(_WIN32)
-#   include "platform_win.h"
 #   include "mmsystem.h"
+#   include "platform_win.h"
```
Comment (ZhilkinSerg) [inline src/translations.cpp:53]: This one should not be reordered and come before previous header.

File: src/translations.cpp [hunk 10]
```diff
@@ -42,12 +45,12 @@ static void reload_names()
 static bool sanity_checked_genders = false;
 
 #if defined(LOCALIZE)
-#include "options.h"
 #include "debug.h"
+#include "options.h"
 #include "ui.h"
 #if defined(_WIN32)
-#   include "platform_win.h"
 #   include "mmsystem.h"
```
Comment (ZhilkinSerg) [inline src/translations.cpp:52]:
```suggestion
#   include "platform_win.h"
```

File: src/translations.cpp (hunk 9 above)
Comment (ZhilkinSerg) [inline src/translations.cpp:53]:
```suggestion
#   include "v.h"
```

**Comment by ZhilkinSerg** [comment]
All build variants passed - there is an unrelated test failure which can be ignored for now.

**Comment by jbytheway** [comment]
Thanks for all the help pushing this through :).

**Pull request merged**